dotnet/server/src/Azure.Mobile.Server/TableController.cs

```diff
@@ -104,6 +104,15 @@ public ITableRepository<TEntity> TableRepository
         /// <returns>The prepared item</returns>
         public virtual TEntity PrepareItemForStore(TEntity item) => item;
 
+        /// <summary>
+        /// Prepares the item for storing into the backend store.  This is a opportunity for the application
+        /// to add additional meta-data that is not passed back to the client (such as the user ID in a 
+        /// personal data store).
+        /// </summary>
+        /// <param name="item">The item to be prepared</param>
+        /// <returns>The prepared item</returns>
+        public virtual Task<TEntity> PrepareItemForStoreAsync(TEntity item)
+            => Task.Run(() => PrepareItemForStore(item));
```

Comment:
```suggestion
            => Task.FromResult(PrepareItemForStore(item));
```
I think here we could return a completed task with the result of the execution of the sync method instead of moving the work to the thread pool.